### DISCUSSION TITLE
SOTA functions

### DISCUSSION THREAD

**Discussion by tuxintrouble** [magicbug/Cloudlog#2956] [Q&A] [answered]
Hi!
I have logged SOTA QSOs in Cloudlog. To do so, I have created a new station for the summit and entered the summit reference there. I have not entered a summit reference with the QSOs as they where not S2S, but made sure, I used the newly created station as active station for the QSOs.

Now when I try to export SOTA csv logs, only two QSOs are included in the export, even through the others include the SOTA reference though the staton.

Also Awards-SOTA does not show any entries. What is the expected behavior there?

Thanks
Seb

**Comment by phl0** [comment]
I think I made a mistake here. IIRC I implemented only looking for DXes SOTA ref. Will have a look.

**Comment by phl0** [comment]
Just did a few tests here. Created a station location, configured SOTA summit and made it the active station profile. Then logged a QSO without a SOTA reference for the worked station. Then did a SOTA csv export and the QSO has been exported. So it works as it should imho. To debug your case we would need a little for feedback then. Did you use the SOTA export function? Did you select the previously created station location upon SOTA export?

**Reply by tuxintrouble** [reply]
Yes, to both.
I just tried again, and it appears, that the resulting csv file contains only the first logged qso for every band. 

Also the Awards-SOTA view (nothing to do with export, I know) is completely empty.

**Reply by phl0** [reply]
Thats weird. I just logged another QSO and now I have two in the resulting csv:

![Screenshot from 2024-02-05 08-29-56](https://github.com/magicbug/Cloudlog/assets/7112907/1e0e00a8-de75-4c55-a571-d0a839005052)

Re awards: Per se it shown only the worked summit references.

**Comment by tuxintrouble** [comment]
Hmm. That is weird. 
I will have a look at the SQL records and see if there are any differences between the records exported and the others, later.

> Re awards: Per se it shown only the worked summit references.

So it is mostly meant for chasers or s2s?

**Comment by phl0** [comment]
> Hmm. That is weird. I will have a look at the SQL records and see if there are any differences between the records exported and the others, later.

Yeah that would be good for further insight. Looking forward to your results.

> > Re awards: Per se it shown only the worked summit references.
> 
> So it is mostly meant for chasers or s2s?

Jepp. Was implemented for that. But free to be extended :)

**Reply by tuxintrouble** [reply]
OK, Here is what happens:

The field TABLE_HRD_CONTACTS_V01.COL_MY_SOTA_REF is copied over from the "station.station_sota" at the creation of the log entry only. 
If the qso is created with another station profile, and the log entry is edited to change the station profile later, COL_MY_SOTA_REF is not updated to the value of the new station profile.

So the bug might be that not all fields of the station profile are updated during the change / edit of an existing qso. It could be that even more fields are affected by this in the update function - IDK.
I can't check / fix as I am not PHP savvy enough to know what I am doing ;-) .
I hope this helps anyways.

Can you CFM?
Shall I post an issue?

**Reply by phl0** [reply]
I see. So the issue lies with QSOs being assigned to a SOTA station location later?

**Comment by tuxintrouble** [comment]
> I see. So the issue lies with QSOs being assigned to a SOTA station location later?

Well... The SOTA reference in a logged qso is not a relation to the SOTA reference in the station, but its value is copied from the station profile at creation time of the log entry. When the log entry is edited later and a different station profile is assigned, the station is referenced correctly in the log entry, but the log entries field for the sota reference is not updated. The function for editing the log entry has to be changed so that not only the id of the station profile gets changed, but the field COL_MY_SOTA_REF of the log entry gets updated with the SOTA reference information from the newly assigned station profile. Basically the exact same way as it is done at the creation of a new log entry.
Does that make sense?

**Comment by phl0** [comment]
> > I see. So the issue lies with QSOs being assigned to a SOTA station location later?
> 
> Does that make sense?

It does. Looking into it.

**Comment by phl0** [comment, marked as answer]
I made a patch here: https://github.com/magicbug/Cloudlog/pull/2958. Would you be able to test this?

**Reply by tuxintrouble** [reply]
I tested for SOTA only, but this seems to work perfectely fine.
Thanks for the quick fix!
